x-pack/plugins/security_solution/server/ui_settings.ts

```diff
@@ -322,6 +323,30 @@ export const initUiSettings = (
       requiresPageReload: true,
       schema: schema.arrayOf(schema.string()),
     },
+    [EXCLUDED_DATA_TIERS_FOR_RULE_EXECUTION]: {
+      name: i18n.translate(
+        'xpack.securitySolution.uiSettings.excludedDataTiersForRuleExecutionLabel',
+        {
+          defaultMessage: 'Exclude cold or frozen data tier from rule execution',
+        }
+      ),
+      description: i18n.translate(
+        'xpack.securitySolution.uiSettings.excludedDataTiersForRuleExecutionDescription',
+        {
+          defaultMessage: `
```

Comment:
@nastasha-solomon, can you have a look at this advanced settings name and description?
Let us know, if we can improved it

Thanks

Reply:
@vitaliidm, what do you think about aligning this advanced setting's name with the advanced setting that excludes cold and frozen data tiers in Analyzer (i.e., change `excludedDataTiersForRuleExecution` to `excludeColdAndFrozenTiersInRules` or `excludeSpecifiedTiersInRules` )?

<img width="600" alt="Screenshot 2024-07-24 at 9 10 56 AM" src="https://github.com/user-attachments/assets/c3359965-010e-4ea3-b838-f338ce514f3a">

I also really like that the `excludeColdAndFrozenTiersInAnalyzer`description explains what the setting does, rather than when you'd use it. There are n number of use cases for toggling a setting that includes/excludes data in a query, so it might be more practical to inform user about what will happen if they use the setting instead of why it's useful for them. Here's a potential revision to consider:

When enabled, events from the specified data tiers are not searched during rules executions. If you specify multiple data tiers, separate values with commas. For example: data_frozen,data_cold 


If you wanted to keep the bit about the value that the setting's offering, here's an alternative:

When enabled, events from the specified data tiers are not searched during rules executions. This might help to improve rule performance or reduce execution time. If you specify multiple data tiers, separate values with commas. For example: data_frozen,data_cold

Reply:
> what do you think about aligning this advanced setting's name with the advanced setting that excludes cold and frozen data tiers in Analyzer (i.e., change excludedDataTiersForRuleExecution to excludeColdAndFrozenTiersInRules or excludeSpecifiedTiersInRules )?

@stash, I think `excludedDataTiersForRuleExecution` suits better for the purposes. Since Analyzer is boolean switch and this option is string of values - so its name reflects that data tiers in that config would be excluded during rule execution. Initially, users were able to exclude any tier(hit and warm too), but we decided to limit it just to frozen and cold. This name would also help, if decide to include warm tier there too at some point.


> When enabled, events from the specified data tiers are not searched during rules executions. This might help to improve rule performance or reduce execution time. If you specify multiple data tiers, separate values with commas. For example: data_frozen,data_cold

I like this option. Only change I would put there is

> When **configured**, events from the specified data tiers are not searched during rules executions. This might help to improve rule performance or reduce execution time. If you specify multiple data tiers, separate values with commas. For example: data_frozen,data_cold

replacing enabled with configured, since it is not boolean switch that can be enabeld/disabled, but config stringm that can have multiple values

Reply:
@nastasha-solomon 

here is a screenshot with changes
<img width="2268" alt="Screenshot 2024-07-24 at 15 42 38" src="https://github.com/user-attachments/assets/bbfdb60e-9e2c-4195-8dd8-2db2ef2e5336">